pkg/pdtime/clock.go

```diff
@@ -28,41 +28,42 @@ import (
 
 const pdTimeUpdateInterval = 200 * time.Millisecond
 
-type TimeAcquirer interface {
-	// Run run the TimeAcquirer
-	Run(ctx context.Context)
-	// CurrentTimeFromCached returns current time from cache
-	CurrentTimeFromCached() (time.Time, error)
-	// Stop stops the TimeAcquirer
-	Stop()
+// Clock is a time source of PD cluster.
+type Clock interface {
+	// CurrentTime returns current time from PD.
+	CurrentTime() (time.Time, error)
 }
 
-// TimeAcquirerImpl cache time get from PD periodically and cache it
-type TimeAcquirerImpl struct {
-	pdClient  pd.Client
-	timeCache time.Time
-	mu        sync.RWMutex
-	cancel    context.CancelFunc
-	err       error
+// PDClock cache time get from PD periodically and cache it
+type PDClock struct {
+	pdClient pd.Client
+	mu       struct {
+		sync.RWMutex
+		timeCache time.Time
+		err       error
+	}
+	cancel context.CancelFunc
+	stopCh chan struct{}
```

Comment:
why we need this `stopCh`?

Reply:
Nice catch, I want to synchronize `Stop`.

Reply:
It seems that `Clock.Stop()` is not thread-safety, will it cause any possible problem?

Reply:
> It seems that `Clock.Stop()` is not thread-safety, will it cause any possible problem?

I am wrong about this, Never mind.